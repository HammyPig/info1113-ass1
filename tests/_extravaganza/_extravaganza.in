author Homer
genre Fantasy
list all
commands
number copies
add collection typo
add book books.csv 111119
add book books.csv 111119
add book books.csv 199999
add book books.csv 111117
list ALL
aDD COLLECTION books.csv
add collection duplicates.csv
add collection books2.csv
list all
number copies
book 111119 long
add member me
rent 100000 111111
add member you
rent 100001 111111
relinquish all 100000
rent 100001 111111
add member us
rent 100000 111111
rent 100000 111112
rent 100000 111113
rent 100000 111114
member books 100000
relinquish 100000 111111
member books 100000
relinquish all 100000
member books 100000
member history 100000
member history 100001
rent 100001 111112
rent 100001 111113
rent 100001 111114
relinquish all 100001
save collection test_output.csv
relinquish all 100001
rent 100001 111115
rent 100001 111116
rent 100001 111117
rent 100001 111118
rent 100001 111119
rent 100001 111120
relinquish all 100001
member history 100001
rent 100002 111112
rent 100002 111113
rent 100002 111114
cOMMON 100001 100002 100003 100004
cOMMON 100001 100002 100003
relinquish all 100002
common 100001 100002
common 100000 100002 100001
common 100002 100001 100000
author Homer
genre Fantasy
genre horror movies
author james walden
list available long
list all long
add member its a me
common 100000 100001 100002 100003
exIt