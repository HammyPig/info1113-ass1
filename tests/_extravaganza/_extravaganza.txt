checks everything (oh and also checks if exit works)