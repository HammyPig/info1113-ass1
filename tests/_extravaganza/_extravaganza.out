user: No books in system.

user: No books in system.

user: No books in system.

user: EXIT ends the library process
COMMANDS outputs this help string

LIST ALL [LONG] outputs either the short or long string for all books
LIST AVAILABLE [LONG] outputs either the short of long string for all available books
NUMBER COPIES outputs the number of copies of each book
LIST GENRES outputs the name of every genre in the system
LIST AUTHORS outputs the name of every author in the system

GENRE <genre> outputs the short string of every book with the specified genre
AUTHOR <author> outputs the short string of every book by the specified author

BOOK <serialNumber> [LONG] outputs either the short or long string for the specified book
BOOK HISTORY <serialNumber> outputs the rental history of the specified book

MEMBER <memberNumber> outputs the information of the specified member
MEMBER BOOKS <memberNumber> outputs the books currently rented by the specified member
MEMBER HISTORY <memberNumber> outputs the rental history of the specified member

RENT <memberNumber> <serialNumber> loans out the specified book to the given member
RELINQUISH <memberNumber> <serialNumber> returns the specified book from the member
RELINQUISH ALL <memberNumber> returns all books rented by the specified member

ADD MEMBER <name> adds a member to the system
ADD BOOK <filename> <serialNumber> adds a book to the system

ADD COLLECTION <filename> adds a collection of books to the system
SAVE COLLECTION <filename> saves the system to a csv file

COMMON <memberNumber1> <memberNumber2> ... outputs the common books in members' history

user: No books in system.

user: No such collection.

user: Successfully added: Harry Potter 2 (J.K. Rowling).

user: Book already exists in system.

user: No such book in file.

user: Successfully added: Batman Volume 1 (Bob Kane).

user: Harry Potter 2 (J.K. Rowling)
Batman Volume 1 (Bob Kane)

user: 15 books successfully added.

user: No books have been added to the system.

user: No such collection.

user: Harry Potter 2 (J.K. Rowling)
Batman Volume 1 (Bob Kane)
To Kill a Mockingbird (Harper Lee)
The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)
The Hitchhiker's Guide to the Galaxy (Douglas Adams)
So Long and Thanks for all the Fish (Douglas Adams)
Harry Potter 1 (J.K. Rowling)
Harry Potter 3 (J.K. Rowling)
Harry Potter 4 (J.K. Rowling)
Harry Potter 5 (J.K. Rowling)
Harry Potter 6 (J.K. Rowling)
Wasteland (T.S. Eliot)
Biggles in the Battle (W.E. Johns)
Aeneid (Virgil)
Halloween (Jamie Lee Curtis)

user: Aeneid (Virgil): 1
Batman Volume 1 (Bob Kane): 1
Biggles in the Battle (W.E. Johns): 1
Halloween (Jamie Lee Curtis): 1
Harry Potter 1 (J.K. Rowling): 1
Harry Potter 2 (J.K. Rowling): 1
Harry Potter 3 (J.K. Rowling): 1
Harry Potter 4 (J.K. Rowling): 1
Harry Potter 5 (J.K. Rowling): 1
Harry Potter 6 (J.K. Rowling): 1
So Long and Thanks for all the Fish (Douglas Adams): 1
The Hitchhiker's Guide to the Galaxy (Douglas Adams): 1
The Hobbit (J.R.R. Tolkien): 1
The Iliad (Homer): 1
The Odyssey (Homer): 1
To Kill a Mockingbird (Harper Lee): 1
Wasteland (T.S. Eliot): 1

user: 111119: Harry Potter 2 (J.K. Rowling, Fantasy)
Currently available.

user: Success.

user: Success.

user: Success.

user: Book is currently unavailable.

user: Success.

user: Success.

user: Success.

user: Book is currently unavailable.

user: Success.

user: Success.

user: Success.

user: The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)

user: Unable to return book.

user: The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)

user: Success.

user: Member not currently renting.

user: To Kill a Mockingbird (Harper Lee)
The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)

user: No rental history for member.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: Success.

user: To Kill a Mockingbird (Harper Lee)
The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)
The Hitchhiker's Guide to the Galaxy (Douglas Adams)
So Long and Thanks for all the Fish (Douglas Adams)
Batman Volume 1 (Bob Kane)
Harry Potter 1 (J.K. Rowling)
Harry Potter 2 (J.K. Rowling)
Harry Potter 3 (J.K. Rowling)

user: Success.

user: Success.

user: Success.

user: No such member in system.

user: No such member in system.

user: Success.

user: The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)

user: The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)

user: The Hobbit (J.R.R. Tolkien)
The Iliad (Homer)
The Odyssey (Homer)

user: The Iliad (Homer)
The Odyssey (Homer)

user: The Hobbit (J.R.R. Tolkien)
Harry Potter 1 (J.K. Rowling)
Harry Potter 2 (J.K. Rowling)
Harry Potter 3 (J.K. Rowling)
Harry Potter 4 (J.K. Rowling)
Harry Potter 5 (J.K. Rowling)
Harry Potter 6 (J.K. Rowling)

user: No books with genre horror movies.

user: No books by james walden.

user: 111119: Harry Potter 2 (J.K. Rowling, Fantasy)
Currently available.

111117: Batman Volume 1 (Bob Kane, Comic)
Currently available.

111111: To Kill a Mockingbird (Harper Lee, Historical Fiction)
Currently available.

111112: The Hobbit (J.R.R. Tolkien, Fantasy)
Currently available.

111113: The Iliad (Homer, Poetry)
Currently available.

111114: The Odyssey (Homer, Poetry)
Currently available.

111115: The Hitchhiker's Guide to the Galaxy (Douglas Adams, Comedy)
Currently available.

111116: So Long and Thanks for all the Fish (Douglas Adams, Comedy)
Currently available.

111118: Harry Potter 1 (J.K. Rowling, Fantasy)
Currently available.

111120: Harry Potter 3 (J.K. Rowling, Fantasy)
Currently available.

111121: Harry Potter 4 (J.K. Rowling, Fantasy)
Currently available.

111122: Harry Potter 5 (J.K. Rowling, Fantasy)
Currently available.

111123: Harry Potter 6 (J.K. Rowling, Fantasy)
Currently available.

111124: Wasteland (T.S. Eliot, Poetry)
Currently available.

111125: Biggles in the Battle (W.E. Johns, War)
Currently available.

111126: Aeneid (Virgil, Epic)
Currently available.

111127: Halloween (Jamie Lee Curtis, Horror)
Currently available.

user: 111119: Harry Potter 2 (J.K. Rowling, Fantasy)
Currently available.

111117: Batman Volume 1 (Bob Kane, Comic)
Currently available.

111111: To Kill a Mockingbird (Harper Lee, Historical Fiction)
Currently available.

111112: The Hobbit (J.R.R. Tolkien, Fantasy)
Currently available.

111113: The Iliad (Homer, Poetry)
Currently available.

111114: The Odyssey (Homer, Poetry)
Currently available.

111115: The Hitchhiker's Guide to the Galaxy (Douglas Adams, Comedy)
Currently available.

111116: So Long and Thanks for all the Fish (Douglas Adams, Comedy)
Currently available.

111118: Harry Potter 1 (J.K. Rowling, Fantasy)
Currently available.

111120: Harry Potter 3 (J.K. Rowling, Fantasy)
Currently available.

111121: Harry Potter 4 (J.K. Rowling, Fantasy)
Currently available.

111122: Harry Potter 5 (J.K. Rowling, Fantasy)
Currently available.

111123: Harry Potter 6 (J.K. Rowling, Fantasy)
Currently available.

111124: Wasteland (T.S. Eliot, Poetry)
Currently available.

111125: Biggles in the Battle (W.E. Johns, War)
Currently available.

111126: Aeneid (Virgil, Epic)
Currently available.

111127: Halloween (Jamie Lee Curtis, Horror)
Currently available.

user: Success.

user: No common books.

user: Ending Library process.
